htlcswitch/link.go

```diff
@@ -509,6 +509,13 @@ func (l *channelLink) Stop() {
 	close(l.quit)
 	l.wg.Wait()
 
+	// Now that the htlcManager has completely exited, reset the packet
+	// courier. This allows the mailbox to revaluate any lingering Adds that
+	// were delivered but didn't make it on a commitment to be failed back
+	// if the link is offline for an extended period of time. The error is
+	// ignored since it can only fail when the daemon is exiting.
+	_ = l.mailBox.ResetPackets()
```

Comment:
Currently we wait to do this until after the channel reest to avoid any race conditions (we cancel back, but then connect to the node and actually commit the HTLC). Still getting through the rest of this diff, but I think we may need stronger synchronization on the commitment and pending commitment level.

Reply:
Actually, I think this is ok, given that we always ack the packets _before_ we transmit any messages to the remote party.